test/Microsoft.AspNetCore.Mvc.FunctionalTests/WebApiCompatShimActionSelectionTest.cs

```diff
@@ -329,6 +327,25 @@ public WebApiCompatShimActionSelectionTest(MvcTestFixture<WebApiCompatShimWebSit
         }
 
         [Theory]
+        [InlineData("api/Admin/Test", "PostUser")]
+        [InlineData("api/Admin/Test?name=mario&age=10", "PostUserByNameAndAge")]
+        public async Task LegacyActionSelection_OverloadedAction_WithUnnamedAction(string requestUrl, string expectedActionName)
```

Comment:
These actually belong to the PR, as discussed with @rynowak I kept the new behavior of returning 415 when no input formatter can be found even in the web api compat shim. As part of that we try to read the body of the request and in order to make the Web API compat shim test pass, I separated the ones that used POST into their own test method and added a dummy body (it's not used on the action) to keep the test happy.

What we care about is the action selection piece of it, so adding a dummy body is just ok to prevent the BodyModelBinder and UnsupportedContentTypeFilter from triggering a 415